optimum/neuron/modeling_base.py

```diff
@@ -316,34 +278,86 @@ def _export(
             "disable_fallback": disable_fallback,
         }
 
-        input_names, output_names = export(
-            model=model,
-            config=neuron_config,
-            output=save_dir_path / NEURON_FILE_NAME,
-            compiler_workdir=compiler_workdir,
-            inline_weights_to_neff=inline_weights_to_neff,
-            optlevel=optlevel,
-            **compiler_kwargs,
-        )
-
-        config = store_compilation_config(
-            config=model.config,
-            input_shapes=input_shapes,
-            compiler_kwargs=compiler_kwargs,
-            input_names=input_names,
-            output_names=output_names,
-            dynamic_batch_size=dynamic_batch_size,
-            compiler_type=compiler_type,
-            compiler_version=compiler_version,
-            inline_weights_to_neff=inline_weights_to_neff,
-            optlevel=optlevel,
-            task=task,
-        )
-
-        config.save_pretrained(save_dir_path)
-        maybe_save_preprocessors(model_id, save_dir_path, src_subfolder=subfolder)
+        if (
+            not inline_weights_to_neff and not disable_neuron_cache and is_neuronx_available()
+        ):  # TODO: support caching of Inf1 as well
+            # Check if the cache exists
+            compilation_config = store_compilation_config(
+                config=config,
+                input_shapes=kwargs_shapes,
+                compiler_kwargs=compiler_kwargs,
+                dynamic_batch_size=dynamic_batch_size,
+                compiler_type=NEURON_COMPILER_TYPE,
+                compiler_version=NEURON_COMPILER_VERSION,
+                inline_weights_to_neff=inline_weights_to_neff,
+                optlevel=optlevel,
+                model_type=getattr(config, "model_type", None),
+                task=task,
+            )
+            cache_config = build_cache_config(compilation_config)
+            cache_entry = ModelCacheEntry(model_id=model_id, config=cache_config)
+            cache_repo_id = load_custom_cache_repo_name_from_hf_home()
+            compile_cache = _create_hub_compile_cache_proxy(cache_repo_id=cache_repo_id)
```

Comment:
You should modify this internal helper to `create_hub_compile_cache_proxy`.
Also, you should not pass cache_repo_id, and let the helper get the default value for you.

Reply:
Great catch, applied!